server/render.js

```diff
@@ -53,62 +76,68 @@ function render(req, res, next) {
       return next();
     }
 
-    const createElement = (Component, props) => (
-      <Component {...props} getCookie={key => req.cookies[key]} />
-    );
-
-    // Todo: render on errors as well
-    const store = configureStore();
-    const app = (
-      <Provider store={store}>
-        <RouterContext {...renderProps} createElement={createElement} />
-      </Provider>
-    );
-
-    const respond = (error = null) => {
-      const helmet = Helmet.rewind();
-      const render = (body = '', state = {}) =>
-        res.send(
-          renderPage({
-            body,
-            state,
-            helmet
-          })
-        );
-
-      // Skip server side rendering on timeout errors:
-      if (error === TimeoutError) {
-        return render();
-      }
-
-      try {
+    Raven.context(function() {
+      const createElement = (Component, props) => (
+        <Component {...props} getCookie={key => req.cookies[key]} />
+      );
+
+      const raven = new UniversalRavenNode(Raven);
+
+      const store = configureStore({}, raven);
+      const app = (
+        <Provider store={store}>
+          <RouterContext {...renderProps} createElement={createElement} />
+        </Provider>
+      );
+
+      const reportError = (error: Error) => {
+        try {
+          // $FlowFixMe
+          const err = error.error ? error.payload : error;
+          log.error(err, 'render_error');
```

Comment:
maybe the log should at least be outside the try catch, so we don't completely ignore the error

Reply:
Yeah. But I think it is nice to guarantee that `reportError` is a nothrow. It is kinda boring if our error reporting function throws an exception, resulting in a timeout for the user.